src/legacy/core_plugins/console/np_ready/public/application/stores/editor.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Reducer } from 'react';
+import { produce } from 'immer';
```

Comment:
TIL about `immer` 😄

Reply:
Haha yeah it's really great 🎉

Reply:
Talking about `immer`, it seems we need to add this dependency into `package.json`, I see it only in `xpack/package.json`, but this is OSS plugin @jloleysens (curious if this code works at all in OSS-only build?)

Reply:
@azasypkin After building with `node ./scripts/build.js --oss --docker --skip-archives --skip-os-packages` and opening console I saw the following:

<img width="1920" alt="Screenshot 2020-01-02 at 11 43 57" src="https://user-images.githubusercontent.com/8155004/71663478-af9d1980-2d55-11ea-8269-752708f56f81.png">

I'll open a fix PR shortly :+1:. Thanks for weighing in here!